nbextensions/bulkImportCell/bulkImportCell.js

```diff
@@ -366,10 +371,40 @@ define([
                         );
                     }
                 }
-                const defaultSpecModel = specs[appId].makeDefaultedModel();
-                otherParamIds[fileType].forEach((paramId) => {
-                    initialParams[fileType].params[paramId] = defaultSpecModel[paramId];
+
+                const inputFileIds = fileParamIds[fileType].filter(
+                    (param) => !outputParamIds[fileType].includes(param)
+                );
+                // if there's an appParameters array, then we need to process that into files
+                // if it's greater than 1 in length, just use the non-file params from the first one
+                // TODO: make a flag that there's more than 1 to show a warning
+                const appParameters = typesToFiles[fileType].appParameters || [];
+                appParameters.forEach((appParamSet) => {
+                    // store all the "input files" in the files array for that type
+                    inputFileIds.forEach((paramId) => {
+                        typesToFiles[fileType].files.push(appParamSet[paramId]);
+                    });
+                    // make a reduction with just the file paths / outputs (fileParamIds) for each parameter
+                    const filePathRow = fileParamIds[fileType].reduce((fpRow, paramId) => {
+                        fpRow[paramId] = appParamSet[paramId];
+                        return fpRow;
+                    }, {});
+                    initialParams[fileType].filePaths.push(filePathRow);
                 });
+
+                const otherParams = specs[appId].makeDefaultedModel();
```

Comment:
Keep using the default values for the given app id. But if we have the appParameters array, we should just use the first item and override the defaults with them.